lms/djangoapps/program_enrollments/api/v1/tests/test_views.py

```diff
@@ -0,0 +1,307 @@
+        self.course = self.program["courses"][0]
+        self.course_run = self.course["course_runs"][0]
+        self.course_key = CourseKey.from_string(self.course_run["key"])
+        CourseOverviewFactory(id=self.course_key)
+        self.course_not_in_program = CourseFactory()
+        self.course_not_in_program_key = CourseKey.from_string(
+            self.course_not_in_program["course_runs"][0]["key"]
+        )
+        CourseOverviewFactory(id=self.course_not_in_program_key)
+        self.path_suffix = self.build_path(self.program_uuid, self.course_run["key"])
+
+    def learner_enrollment(self, student_key, status="active"):
+        """
+        Convenience method to create a learner enrollment record
+        """
+        return {"student_key": student_key, "status": status}
+
+    def build_path(self, program_uuid, course):
+        """
+        Convenience method to build a path for a program course enrollment request
+        """
+        return 'programs/{}/course/{}/enrollments/'.format(program_uuid, course)
+
+    def create_program_enrollment(self, external_user_key, user=False):
+        """
+        Creates and returns a ProgramEnrollment for the given external_user_key and
+        user if specified.
+        """
+        program_enrollment = ProgramEnrollmentFactory.create(
+            external_user_key=external_user_key,
+            program_uuid=self.program_uuid,
+        )
+        if user is not False:
+            program_enrollment.user = user
+            program_enrollment.save()
+        return program_enrollment
+
+    def test_enrollments(self):
+        self.create_program_enrollment('l1')
+        self.create_program_enrollment('l2')
+        self.create_program_enrollment('l3', user=None)
+        self.create_program_enrollment('l4', user=None)
+        post_data = [
+            self.learner_enrollment("l1", "active"),
+            self.learner_enrollment("l2", "inactive"),
+            self.learner_enrollment("l3", "active"),
+            self.learner_enrollment("l4", "inactive"),
+        ]
+        response = self.post(self.path, post_data, self.admin_user)
+        self.assertEqual(200, response.status_code)
+        self.assertDictEqual(
+            {
+                "l1": "active",
+                "l2": "inactive",
+                "l3": "active",
+                "l4": "inactive",
+            },
+            response.data
+        )
+        self.assert_program_course_enrollment("l1", "active", True)
+        self.assert_program_course_enrollment("l2", "inactive", True)
+        self.assert_program_course_enrollment("l3", "active", False)
+        self.assert_program_course_enrollment("l4", "inactive", False)
+
+    def assert_program_course_enrollment(self, external_user_key, expected_status, has_user):
+        """
+        Convenience method to assert that a ProgramCourseEnrollment has been created,
+        and potentially that a CourseEnrollment has also been created
+        """
```

Comment:
I think we should also assert `enrollment.program_uuid` and `enrollment.course_enrollment.course_key` are correct.